src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -461,34 +468,48 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
             }
             Debug.Assert(!rosPointerCtor.HasUnsupportedMetadata);
 
-            ImmutableArray<byte> data = default;
             ArrayTypeSymbol? arrayType = null;
             TypeSymbol? elementType = null;
-            if (wrappedExpression is BoundArrayCreation ac)
+            if (wrappedExpression is not BoundArrayCreation { InitializerOpt: { } initializer } ac)
             {
-                // Get the array type and its element type.
-                arrayType = (ArrayTypeSymbol)ac.Type;
-                elementType = arrayType.ElementType;
+                return false;
+            }
 
-                // This optimization is only supported for core primitive types that can be stored in metadata blobs.
-                // For enums, we need to use the underlying type.
-                specialElementType = elementType.EnumUnderlyingTypeOrSelf().SpecialType;
-                if (!IsTypeAllowedInBlobWrapper(specialElementType))
-                {
-                    return false;
-                }
+            // Get the array type and its element type.
+            arrayType = (ArrayTypeSymbol)ac.Type;
+            elementType = arrayType.ElementType;
+
+            ImmutableArray<BoundExpression> initializers = initializer.Initializers;
+            var elementCount = initializers.Length;
+            if (elementCount == 0)
+            {
+                emitEmptyReadonlySpan(spanType, wrappedExpression, used, inPlaceTarget);
+                return true;
+            }
 
-                // Get the data and number of elements that compose the initialization.
-                elementCount = TryGetRawDataForArrayInit(ac.InitializerOpt, out data);
+            if (IsPeVerifyCompatEnabled())
```

Comment:
>IsPeVerifyCompatEnabled [](http://example.com/codeflow?start=16&length=23)

I know, I asked to move this 'if' here. But now I realized, that the code path utilizing ```tryEmitAsCachedArrayOfConstants``` probably is not subject for verifiability issue. Consider if it would make sense to move this `if` again, right before the ```ImmutableArray<byte> data = GetRawDataForArrayInit(initializers);``` line.